### PR TITLE
Linux sensors update for reading temperature

### PR DESCRIPTION
**Pull request by cweitkamp**
- added additional path for reading temperature

Signed-off-by: Christoph Weitkamp <github.christophweitkamp.de>

### CODE REVIEW

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java [hunk 1]
```diff
@@ -138,7 +171,7 @@ public double getCpuTemperature() {
     public double getCpuVoltage() {
         if (this.hwmonMap.containsKey(VOLTAGE)) {
             String hwmon = this.hwmonMap.get(VOLTAGE);
-            // Should return a single line of millidegrees Celsius
+            // Should return a single line of millivolt
```
Comment (cweitkamp) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java]: Copy & paste error of provious author?

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java [hunk 2]
```diff
@@ -41,30 +41,57 @@
 
     // Base HWMON path, adds 0, 1, etc. to end for various sensors
     private static final String HWMON = "/sys/class/hwmon/hwmon";
+    // Base THERMAL_ZONE path, adds 0, 1, etc. to end for temperature sensors
+    private static final String THERMAL_ZONE = "/sys/class/thermal/thermal_zone";
 
     // Map from sensor to path
     private Map<String, String> hwmonMap = new HashMap<>();
```
Comment (cweitkamp) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java]: I want to rename this property - `hwmonMap` - to `sensorsMap`. It sounds a little bit more related than the current name. I also want to remove the use of `this`'s. May I?

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java (hunk 1 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java]: The previous author thinks so. ;)

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java (hunk 2 above)
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java]: No objections.  Trying to remember why I added `this` everywhere... I think some automated code review tool recommended it as a defensive practice.

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java [hunk 3]
```diff
@@ -41,30 +41,57 @@
...
-                });
-                if (matchingFiles != null && matchingFiles.length > 0) {
-                    this.hwmonMap.put(sensor, String.format("%s/%s", path, sensor));
+                @Override
+                public boolean accept(File f) {
+                    return f.getName().startsWith(sensorPrefix) && f.getName().endsWith("_input");
+                }
+            });
+        }
+        // Iterate over all thermal_zone* directories and look for sensor files
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSensors.java]: Add to this comment that we only do this if we didn't already find the temp

**Comment by dbwiddis** [comment]
Looks like the build is failing on openjdk.  It can be worked around (or probably ignored since I expect they'll re-fix it), see https://github.com/travis-ci/travis-ci/issues/5227

**Pull request merged**